src/server/protocol.ts

```diff
@@ -1106,9 +1106,9 @@ namespace ts.server.protocol {
         options: ts.CompilerOptions;
 
         /**
-         * true if project language service is disabled
+         * Filename of the last file analyzed before disabling the language service. undefined, if the language service is enabled.
          */
-        languageServiceDisabled: boolean;
+        exceededFilename: string | undefined;

```

Comment:
The name of the property is not great as it doesn't specify that this is means the language service is disabled. I open to suggestions.

Reply:
```lastFileAnalysedToDisableLanguageService``` ?
Removing ```languageServiceDisabled``` from here would be breaking change...

Reply:
I think is to long, better lastFileExceededProgramSize?
I'm aware of the breaking change and working on it. My plan is to support both properties for backward compatibility.

Reply:
Added back for backward compatibility